osquery/core/tables.cpp

```diff
@@ -197,5 +135,88 @@ std::string TablePlugin::statement() const {
   return "CREATE TABLE " + name_ + columnDefinition();
 }
 
+RouteInfo TablePlugin::routeInfo() const {
+  return {{"definition", columnDefinition()}};
+}
+
+bool ConstraintList::matches(const std::string& expr) const {
+  // Support each SQL affinity type casting.
+  if (affinity == "TEXT") {
+    return literal_matches<TEXT_LITERAL>(expr);
+  } else if (affinity == "INTEGER") {
+    INTEGER_LITERAL lexpr = AS_LITERAL(INTEGER_LITERAL, expr);
+    return literal_matches<INTEGER_LITERAL>(lexpr);
+  } else if (affinity == "BIGINT") {
+    BIGINT_LITERAL lexpr = AS_LITERAL(BIGINT_LITERAL, expr);
+    return literal_matches<BIGINT_LITERAL>(lexpr);
+  } else if (affinity == "UNSIGNED_BIGINT") {
+    UNSIGNED_BIGINT_LITERAL lexpr = AS_LITERAL(UNSIGNED_BIGINT_LITERAL, expr);
+    return literal_matches<UNSIGNED_BIGINT_LITERAL>(lexpr);
+  } else {
+    // Unsupprted affinity type.
+    return false;
+  }
+}
+
+template <typename T>
+bool ConstraintList::literal_matches(const T& base_expr) const {
+  bool aggregate = true;
+  for (size_t i = 0; i < constraints_.size(); ++i) {
+    T constraint_expr = AS_LITERAL(T, constraints_[i].expr);
+    if (constraints_[i].op == EQUALS) {
+      aggregate = aggregate && (base_expr == constraint_expr);
+    } else if (constraints_[i].op == GREATER_THAN) {
+      aggregate = aggregate && (base_expr > constraint_expr);
+    } else if (constraints_[i].op == LESS_THAN) {
+      aggregate = aggregate && (base_expr < constraint_expr);
+    } else if (constraints_[i].op == GREATER_THAN_OR_EQUALS) {
+      aggregate = aggregate && (base_expr >= constraint_expr);
+    } else if (constraints_[i].op == LESS_THAN_OR_EQUALS) {
+      aggregate = aggregate && (base_expr <= constraint_expr);
+    } else {
+      // Unsupported constraint.
+      return false;
+    }
+    if (!aggregate) {
+      // Speed up comparison.
+      return false;
+    }
+  }
+  return true;
+}
+
+std::set<std::string> ConstraintList::getAll(ConstraintOperator op) const {
```

Comment:
hmm, the operator is an enum, not sure if passing the reference saves much :/ int might be smaller, ha!